vision/src/autogluon/vision/detector/detector.py

```diff
@@ -238,20 +238,27 @@ def fit(self,
         if ngpus_per_trial is not None:
             config['ngpus_per_trial'] = ngpus_per_trial
         if isinstance(hyperparameters, dict):
-            if 'batch_size' in hyperparameters:
-                bs = hyperparameters['batch_size']
-                if ngpus_per_trial is not None and ngpus_per_trial > 1 and bs > 16:
-                    # using gpus, check batch size vs. available gpu memory
-                    free_gpu_memory = get_gpu_free_memory()
-                    if not free_gpu_memory:
-                        warnings.warn('Unable to detect free GPU memory, we are unable to verify '
-                                      'whether your data mini-batches will fit on the GPU for the specified batch_size.')
-                    elif len(free_gpu_memory) < ngpus_per_trial:
-                        warnings.warn(f'Detected GPU memory for {len(free_gpu_memory)} gpus but {ngpus_per_trial} is requested.')
-                    elif sum(free_gpu_memory[:ngpus_per_trial]) / bs < 1280:
-                        warnings.warn(f'batch_size: {bs} is potentially larger than what your gpus can support ' +
-                                      f'free memory: {free_gpu_memory[:ngpus_per_trial]} ' +
-                                      'Try reducing "batch_size" if you encounter memory issues.')
+            try:
+                if 'batch_size' in hyperparameters:
+                    bs = hyperparameters['batch_size']
+                    if isinstance(bs, Categorical):
+                        bs = max(bs.data)
+                    if isinstance(bs, (Real, Int)):
+                        bs = bs.upper
+                    if ngpus_per_trial is not None and ngpus_per_trial > 1 and bs > 4:
+                        # using gpus, check batch size vs. available gpu memory
+                        free_gpu_memory = get_gpu_free_memory()
+                        if not free_gpu_memory:
+                            warnings.warn('Unable to detect free GPU memory, we are unable to verify '
+                                          'whether your data mini-batches will fit on the GPU for the specified batch_size.')
+                        elif len(free_gpu_memory) < ngpus_per_trial:
+                            warnings.warn(f'Detected GPU memory for {len(free_gpu_memory)} gpus but {ngpus_per_trial} is requested.')
+                        elif sum(free_gpu_memory[:ngpus_per_trial]) / bs < 1280:
```

Comment:
is 1280 the correct number? In image_prediction it is 128. It might make sense to move this into a function and have  both detector and predictor call it to avoid code duplication.